File: Solutions/454. 4Sum II/454. 4Sum II.py
class Solution:
    def fourSumCount(self, A: List[int], B: List[int], C: List[int], D: List[int]) -> int:
        res,dic=0,{}
        for a in A:
            for b in B:
                if a+b in dic:
                    dic[a+b]+=1
                else:
                    dic[a+b]=1
        for c in C:
            for d in D:
                if -c-d in dic:
                    res+=dic[-c-d]
        return res

File: Solutions/454. 4Sum II/454. 4Sum II.java
class Solution {
    public int fourSumCount(int[] A, int[] B, int[] C, int[] D) {
        int res=0;
        var map=new HashMap<Integer,Integer>();
        for(var a:A)
            for(var b:B){
                if(map.containsKey(a+b))
                    map.put(a+b,map.get(a+b)+1);
                else
                    map.put(a+b,1);
                }
        for(var c:C)
            for(var d:D){
                if(map.containsKey(-c-d))
                    res+=map.get(-c-d);
            }
        return res;
    }
}